sympy/core/expr.py

```diff
@@ -3787,7 +3787,12 @@ def unchanged(func, *args):
     False
     """
     f = func(*args)
-    return f.func == func and f.args == tuple([sympify(a) for a in args])
```

Comment:
what happens if you don't use sympify?

Reply:
...`f.args == args`

Reply:
Then, in case of `Piecewise`, `a` will be a `tuple` instead of `Tuple` and `args[i]` will be `ExprCondPair` so that will still not work.

Reply:
It should work. `unchanged` is not as strict as `_aresame`:
```
>>> assert Piecewise((1,x>2)).args == ((1, x>2),)
```

Reply:
Okay, let me try it out.

Reply:
I added an example to the docstring, too.